librb/src/openssl.c

```diff
@@ -413,13 +423,41 @@ rb_setup_ssl_server(const char *const certfile, const char *keyfile,
 		}
 	}
 
-	if(SSL_CTX_set_cipher_list(ssl_ctx_new, cipherlist) != 1)
+
+	int ret1 = SSL_CTX_set_cipher_list(ssl_ctx_new, cipherlist_old);
+
+	if (ret1 != 1)
+	{
+		(void) rb_lib_log("%s: no valid old-style ciphersuites found "
+		                  "in ssl_cipher_list; will use defaults",
+		                  __func__);
+
+		ret1 = SSL_CTX_set_cipher_list(ssl_ctx_new, rb_default_ciphers_old);
+	}
+
+	#ifndef LRB_HAVE_TLS13
+	int ret2 = 0;
+	#else
+	int ret2 = SSL_CTX_set_ciphersuites(ssl_ctx_new, cipherlist_new);
+
+	if (ret2 != 1)
 	{
-		rb_lib_log("%s: SSL_CTX_set_cipher_list: could not configure any ciphers", __func__);
+		(void) rb_lib_log("%s: no valid new-style ciphersuites found "
+		                  "in ssl_cipher_list; will use defaults",
+		                  __func__);
+
+		ret2 = SSL_CTX_set_ciphersuites(ssl_ctx_new, rb_default_ciphers_new);
+	}
+	#endif
+
+	if (ret1 != 1 && ret2 != 1)
+	{
+		rb_lib_log("%s: could not configure any ciphers", __func__);
```

Comment:
Are you able to hit this in tests? Should it say which of old/new failed?

It looks to me like this path is inevitable if !defined(LRB_HAVE_TLS13)

Reply:
The if tests above it already say which of old and/or new failed.